### PR TITLE
Disable hostile ai on protected grids

### PR DESCRIPTION
**Pull request by GreaseMonk**
## About the PR

Disabled hostile ai on protected grids like transport shuttle and frontier outpost

## Why / Balance

Hostile mobs seem to get on board protected areas way too often and kills ssd players mostly, leaving a mess.

## How to test

Spawn any hostile AI that may attack things on frontier and check if it just does nothing, as intended.
Also spawn other AI like chicken, cow, CDET should still work too.

I tested this by dragging AI from a protected grid to a normal grid, which then reactivates the AI again, which may come as a suprise but if players push them out of the docks of frontier, it 'should'  not hit the player - but it might ? . Either way better than what it was.

## Media

https://github.com/user-attachments/assets/9b90d632-ebca-4665-8b65-755bca881550

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

No breaking changes.

**Changelog**

:cl:
- add: Hostile AI no longer work on protected zones.

### CODE REVIEW

**Review by whatston3 — COMMENTED** [review]
Seems like a bit much to me, and open to some exploitation.  I'll defer to others, but doesn't seem critical.

Left a recommendation to remove the mob and all of its contents, Dvir had something like this for the salvage mob restrictions off the top of my head.

File: Content.Server/NPC/HTN/HTNSystem.cs [hunk 1]
```diff
@@ -168,6 +173,17 @@ public void UpdateNPC(ref int count, int maxUpdates, float frameTime)
...
+            // Frontier: Disable hostile AI in pacified zones
+            var grid = Transform(uid).GridUid;
+            if (grid != null && EntityManager.TryGetComponent<ProtectedGridComponent>(grid, out _))
+            {
+                if (EntityManager.TryGetComponent<NpcFactionMemberComponent>(uid, out var npcFactionMemberComponent))
+                {
+                    if (_npcFaction.IsFactionHostile("NanoTrasen", (uid, npcFactionMemberComponent)))
+                        continue;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (GreaseMonk) [reply Content.Server/NPC/HTN/HTNSystem.cs:185]: Excellent idea. i will change this soonish

**Comment by GreaseMonk** [comment]
forgot about this PR.. rip, ill be on it

**Comment by GreaseMonk** [comment]
https://github.com/user-attachments/assets/dd3cf151-83c9-4e80-a3e2-9547f9c630a1

Final result.... should be done

File: Content.Server/NPC/HTN/HTNSystem.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (GreaseMonk) [reply Content.Server/NPC/HTN/HTNSystem.cs]: Done

**Comment by whatston3** [comment]
https://github.com/new-frontiers-14/frontier-station-14/compare/fs-28-disable-hostile-ai-on-protected-grids...whatston3:frontier-station-14:2024-12-06-hostile-ai-on-grid-suggestions?expand=1

Good overall, suggestions above - do think the new kill hostile mob field should default to false.  Added this, applied it to Frontier, cleaned up some unused dependencies, and left comments on others added above.

**Comment by whatston3** [comment]
On further review, I wrote a small system and hooked into the parent changed event for ActiveNPCs.  Should run a lot less frequently, happy with the state of this.

Do give it a read when you can, pushed the changes to the branch (sorry about that), but I think this is good.

**Review by whatston3 — APPROVED** [review]
Forgot to clean up the HTNSystem, think this is good.  Still works, should be fine for _most_ cases in normal gameplay.

The one thing we could _maybe_ do is run this check on ComponentStartup - if an entity gets spawned on the grid, it won't die.  Not sure what happens if it gets disabled and re-enabled on a new grid, either, but I'd wager it misses the event.

Minor point aside, this should be fine.

**Comment by whatston3** [comment]
Another little bit of cleanup - startup function (angry or inactive mobs die on spawn/wakeup).

Slightly changed the order of the checks, cached Transform to a variable to reuse.

No complaints, this looks good to me.

**Review by whatston3 — APPROVED** [review]

**Comment by dvir001** [comment]
Clug walking into frontier moment

**Pull request merged**